src/Symfony/Component/Validator/Constraints/ContainsValidator.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+class ContainsValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Contains) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Contains');
+        }
+
+        if (!\is_string($constraint->expected) && !\is_array($constraint->expected) && !$constraint->callback) {
+            throw new ConstraintDefinitionException('Either "expected" or "callback" must be specified on constraint Contains');
+        }
+        if (null === $value) {
+            return;
+        }
+
+        if (!\is_string($value)) {
+            throw new UnexpectedValueException($value, 'string');
```

Comment:
I think we can support both strings and arrays since one will just use `strpos` and the other `in_array` (I don't think we should support nested containment for arrays, since it's very specific and can be achieved with `Callback` constraints or `IsTrue` on a valid method)

Reply:
@Pierstoval What do you think about a separate constraint validator for `in_array`?

Reply:
In this case, we need `StringContains`, `StringNotContains`, `InArray` and `NotInArray`. That's 4 constraints. Isn't it a bot too much to create 8 classes instead of 4? 😄

Reply:
Well, the way you expose it indeed, it makes no sense. I didn't say anything about `StringNotContains` and `NotInArray`, which are irrelevant as an example and I agree, they don't make any sense.
In the past, I had to work with similar validators/constraints and it was always confusing: is `Contains` doing `in_array` checks or not.

Reply:
I'm not entirely against the idea of having separated classes 😉 I just thought it would be easier if `Contains` could handle both cases 😄